src/x509.c

```diff
@@ -2371,8 +2377,18 @@ static WOLFSSL_X509_EXTENSION* createExtFromStr(int nid, const char *value) {
             break;
         case NID_subject_alt_name:
         {
-            WOLFSSL_GENERAL_NAMES* gns = wolfSSL_sk_new_null();
+            WOLFSSL_GENERAL_NAMES* gns;
             WOLFSSL_GENERAL_NAME* gn;
+
+            if (wolfSSL_ASN1_STRING_set(&ext->value, value, -1)
+                    != WOLFSSL_SUCCESS) {
+                WOLFSSL_MSG("wolfSSL_ASN1_STRING_set error");
+                goto err_cleanup;
+            }
+            ext->value.type = ASN_DNS_TYPE;
```

Comment:
What if `value` is an IP address? Should we detect that case and use `ASN_IP_TYPE` instead?

Reply:
The test case uses `"DNS:example.com, IP:127.0.0.1",`. Seems our internal code handles it fine. But maybe I should add some better tests? I do see some logic differences for `ASN_IP_TYPE` and `ASN_DNS_TYPE`. Thoughts?